src/Npgsql/NpgsqlMultiHostDataSource.cs

```diff
@@ -305,7 +305,7 @@ internal override async ValueTask<NpgsqlConnector> Get(
 
         var exceptions = new List<Exception>();
 
-        var poolIndex = conn.Settings.LoadBalanceHosts ? GetRoundRobinIndex() : 0;
+        var poolIndex = conn.Settings.LoadBalanceHosts == LoadBalanceHosts.True ? GetRoundRobinIndex() : 0;
```

Comment:
Should it also check for `conn.Settings.LoadBalanceHosts == LoadBalanceHosts.Any`?

Reply:
This part of code is not part of any execution as it was the upstream’s code for load balance host. This change was only made to avoid the build failing because of change in the `Load Balance Hosts` parameter type.